### PR TITLE
Fix lxqt-leave's dialog to not set maximum size as minimum size

### PR DESCRIPTION
**Pull request by paulolieuthier**
That leads to a zero-sized window.

### CODE REVIEW

**Comment by agaida** [comment]
GTM

File: lxqt-leave/main.cpp [hunk 1]
```diff
@@ -106,6 +106,6 @@ int main(int argc, char *argv[])
                 Qt::AlignCenter,
                 dialog.size(),
                 qApp->desktop()->screenGeometry(QCursor::pos())));
-    dialog.setMaximumSize(dialog.minimumSize());
+    dialog.setMaximumSize(dialog.size());
```
Comment (palinek) [inline lxqt-leave/main.cpp]: Wouldn't it be better to use `sizeHint()`? Does `size()` return reliable size if the widget wasn't ever shown?

**Comment by pmattern** [comment]
477faf5 is fixing the problem here. Can't say anything about the question raised by palinek.

In case anybody wonders, here's the problem addressed by this PR (discussed on freenode #lxde yesterday):
After compiling a69de25 against Qt 5.*6* on Arch Linux the GUIs of invocations `lxqt-leave [--foobar]` came up sporadically only if the binary got launched from a virtual terminal or a desktop entry file from panel's main menu or dragged into plugin-quicklaunch. They still came up reliably if launched from the menu of lxqt-runner.

File: lxqt-leave/main.cpp (hunk 1 above)
Comment (paulolieuthier) [inline lxqt-leave/main.cpp]: Yeah, I thought that. And tested. `sizeHint()` returned a smaller size than `size()`. As you can see, `size()` has been used in the previous function call. We can set `minimumSize` to `sizeHint()` and rest assured, do you agree?

Comment (palinek) [inline lxqt-leave/main.cpp]: > As you can see, size() has been used in the previous function call.

...and I think this should be changed also

> We can set minimumSize to sizeHint()

yes

Comment (paulolieuthier) [inline lxqt-leave/main.cpp]: > ...and I think this should be changed also

Even setting `minimumSize` to `sizeHint()` before?

Comment (palinek) [inline lxqt-leave/main.cpp]: > > ...and I think this should be changed also
> Even setting minimumSize to sizeHint() before?

Does it make any difference? (I meant to use the `dialog.sizeHint()` in the `dialog.setGeometry()` call)

Comment (paulolieuthier) [inline lxqt-leave/main.cpp]: `size()`:
size
`sizeHint()`:
sizehint

Comment (palinek) [inline lxqt-leave/main.cpp]: I would rather use the `sizeHint()`... and if we want more spacing, it should be done by the layout spacings in the .ui.

...I will push a commit instantly

**Comment by luis-pereira** [comment]
GTM

**Comment by paulolieuthier** [comment]
@luis-pereira thanks. Just a note: this PR still needed history cleanup (see the fixup commit) and rebasing to master (so that a merge commit wouldn't be needed).

**Pull request merged**